roofit/histfactory/src/ParamHistFunc.cxx

```diff
@@ -210,9 +210,13 @@ RooAbsReal& ParamHistFunc::getParameter( Int_t index ) const {
     _numBinsPerDim = getNumBinsPerDim(_dataVars);
   }
 
-  int i = index / n.yz ;
-  int j = (index % n.y) / n.z;
-  int k = index % (n.yz);
+  // Unravel the index to 3D coordinates. We can't use the index directly,
+  // because in the parameter set the dimensions are ordered in reverse order
+  // compared to the RooDataHist (for historical reasons).
+  const int i = index / n.yz;
+  const int tmp = index - i * n.yz;
```

Comment:
It is not equivalent to 
temp=index%n.yz?

Reply:
Same for k

Reply:
K=(index%n.yz)%n.z

Reply:
Thanks, you are right! The expression for tmp and k could be simplified, which I changed now.